monitor/btcscanner/btc_scanner.go

```diff
@@ -87,14 +91,23 @@ func (bs *BtcScanner) Start() {
 	// the bootstrapping should not block the main thread
 	go bs.Bootstrap()
 
-	bs.BtcClient.MustSubscribeBlocks()
-
 	bs.Started.Store(true)
 	bs.logger.Info("the BTC scanner is started")
 
+	if err := bs.btcNotifier.Start(); err != nil {
+		bs.logger.Errorf("Failed starting notifier")
+		return
+	}
+
+	blockNotifier, err := bs.btcNotifier.RegisterBlockEpochNtfn(nil)
+	if err != nil {
+		bs.logger.Errorf("Failed registering block epoch notifier")
+		return
+	}
+
 	// start handling new blocks
 	bs.wg.Add(1)
-	go bs.blockEventHandler()
+	go bs.blockEventHandler(blockNotifier)
```

Comment:
I was worried about possible racing issue between the `Bootstrap` and `blockEventHandler` threads. They both produce confirmed blocks to the same channel, so they need to be sequencial. I can think of two options:
1. keep them as two separate go routines and use the `Synced` controller to ensure `blockEventHandler` happens after `Bootstrap`
2. combine them into a single goroutine and when creating the notifier

I'm in favor of option (2) in which we probably can remove the `Synced` controller. Nevertheless, after bootstrapping, we should be able to get the best known block and put it in `btcNotifier.RegisterBlockEpochNtfn(bestKnownBlock)`, so that we don't duplicate the chain tip